File: 50. Pow(x, n)/myPow.cpp
class Solution {
public:
    double powerHelper(double base, long power){
        if(power == 0) return 1;
        double half = powerHelper(base , power/2);
        double res = half * half;
        if(power % 2 == 1) res *= base;

        return res;
    }
    double myPow(double x, int n) {
        long N = n;

        if(n < 0){
            x = 1/x;
            N = N * -1;
        }

        return powerHelper(x,N);
    }
};

File: 50. Pow(x, n)/myPow.js
var myPow = function (x, n) {
    let N = n

    if (N < 0) {
        x = 1 / x
        N = -N
    }

    function powerHelper(x, n) {
        if (n === 0) return 1;
        let half = powerHelper(x, Math.floor(n / 2))
        let res = half * half
        if (n % 2 === 1) res *= x
        return res
    }

    return powerHelper(x, N)


    // let res = x ** n
    // return res
};